plugins/catalog-backend/src/catalog/DatabaseCatalog.ts

```diff
@@ -15,29 +15,79 @@
  */
 
 import { NotFoundError } from '@backstage/backend-common';
-import path from 'path';
 import Knex from 'knex';
+import path from 'path';
+import fs from 'fs-extra';
 import { v4 as uuidv4 } from 'uuid';
-import { AddLocationRequest, Component, Catalog, Location } from './types';
+import { Logger } from 'winston';
+import {
+  AddLocationRequest,
+  Catalog,
+  Component,
+  Location,
+  componentShape,
+} from './types';
+import YAML from 'yaml';
+
+async function readLocation(location: Location): Promise<Component[]> {
+  switch (location.type) {
+    case 'file': {
+      let parsed;
+      try {
+        const raw = await fs.readFile(location.target, 'utf8');
+        parsed = YAML.parse(raw);
```

Comment:
Might as well `YAML.parseAllDocuments` right? Extra work to be done for each doc is [here](https://github.com/eemeli/yaml/blob/3a9327fb224801547f656bdbec20e5f288759aed/src/index.js#L51)